python/cudf/cudf/tests/test_spilling.py

```diff
@@ -254,6 +254,7 @@ def reload_options():
     manager = get_global_manager()
     assert isinstance(manager, SpillManager)
     assert manager.statistics.level == 2
+    monkeypatch.delenv("CUDF_SPILL")
```

Comment:
It seems like you'd need to call `reload_options()` again after this environment variable is deleted, or else it won't reset the behavior?
```suggestion
    monkeypatch.delenv("CUDF_SPILL")
    reload_options()
```

Do we want to wrap all the `setenv` calls in a monkeypatch context?
```python
with monkeypatch.context() as m:
    m.setenv("CUDF_SPILL", "on")
    reload_options()
    # ...

reload_options()
```

Reply:
Okay, the first thing was addressed by @vyasr's review. But I still would prefer to wrap with a monkeypatch context where needed, instead of setting and unsetting all the variables.

Reply:
Done 😉